### PR TITLE
AInputCurrency

### PR DESCRIPTION
**Pull request by NKDante**

### CODE REVIEW

File: src/ui/AInputCurrency/compositionAPI/InputEventsAPI.js [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+        const positionToSet = cursorPosition;
+        const splitVal = value.split(decimalDivider.value);
+        const intVal = splitVal[0];
+        const floatVal = splitVal[1];
+        let valWithDivider = intVal.split("");
+        valWithDivider.splice(cursorPosition + 1, 1);
+        valWithDivider = valWithDivider.reverse().join("")
+          .replaceAll(thousandDivider.value, "")
+          .match(/.{1,3}/g).join(thousandDivider.value)
+          .split("").reverse().join("");
```
[the remark this answers is not in the document]
Comment (NKDante) [reply src/ui/AInputCurrency/compositionAPI/InputEventsAPI.js:252]: This one works a little differently unfortunately

File: src/ui/AInputCurrency/compositionAPI/InputEventsAPI.js [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+
+  const handleInput = ($event, _value) => {
+    const value = _value || $event.target.value;
+    const decimalDividerIndex = value.indexOf(decimalDivider.value);
+    const hasDecimalDivider = decimalDividerIndex !== -1;
+    const cursorPosition = inputRef.value.selectionStart;
+    const splitVal = value.split(decimalDivider.value);
+    const setMinusSymbol = splitVal[0] === "-" ? "-" : "";
+    const intVal = Number(splitVal[0].replace(/[^0-9]/g, "")).toString();
+    const floatVal = splitVal[1] ? splitVal[1].substring(0, symbolsAfterDecimalDivider.value) : "";
```
[the remark this answers is not in the document]
Comment (NKDante) [reply src/ui/AInputCurrency/compositionAPI/InputEventsAPI.js:93]: For currency we'll not be using exponential numbers. For int we are removing thousand dividers (example: 199.899.999).  For now we are not using thousand dividers for float part.

File: src/ui/AInputCurrency/AInputCurrency.js [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+    UiClearButtonMixinProps,
+    UiMixinProps,
+  ],
+  props: {
+    controlsType: {
+      type: String,
+      required: false,
+      default: inputNumberPluginOptions.value.propsDefault.controlsType,
+      validator: value => ["plus-minus", "arrows", "none"].indexOf(value) !== -1,
+      // TODO: "arrows",
```
[the remark this answers is not in the document]
Comment (NKDante) [reply src/ui/AInputCurrency/AInputCurrency.js:44]: Not yet

**Review by ragnara76 — APPROVED** [review]
all issues and conversations solved

**Review by ragnara76 — APPROVED** [review]

**Pull request merged**